base/libgit2/libgit2.jl

```diff
@@ -333,10 +333,12 @@ function reset!(repo::GitRepo, committish::AbstractString, pathspecs::AbstractSt
     # do not remove entries in the index matching the provided pathspecs with empty target commit tree
     obj === nothing && throw(GitError(Error.Object, Error.ERROR, "`$committish` not found"))
     try
-        reset!(repo, Nullable(obj), pathspecs...)
+        head = reset!(repo, Nullable(obj), pathspecs...)
+        return head
```

Comment:
what's sketchy about this? Now we have finalizers attached, we could just get rid of the try/finally block altogether.